supporting_scripts/analysis-of-endpoint-connections/src/main/java/de/tum/cit/endpointanalysis/EndpointAnalyzer.java

```diff
@@ -0,0 +1,143 @@
+package de.tum.cit.endpointanalysis;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class EndpointAnalyzer {
+
+    private static String EndpointAnalysisResultPath = "endpointAnalysisResult.json";
+
+    private static final Logger logger = LoggerFactory.getLogger(EndpointAnalyzer.class);
+
+    public static void main(String[] args) {
+        analyzeEndpoints();
+        printEndpointAnalysisResult();
+    }
+
+    /**
+     * Analyzes server side endpoints and matches them with client side REST calls.
+     *
+     * This method reads endpoint and REST call information from JSON files,
+     * compares them to find matching REST calls for each endpoint, and writes
+     * the analysis result to a JSON file. Endpoints without matching REST calls
+     * are also recorded.
+     */
+    private static void analyzeEndpoints() {
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            List<EndpointClassInformation> endpointClasses = mapper.readValue(new File(EndpointParser.ENDPOINT_PARSING_RESULT_PATH),
+                    new TypeReference<List<EndpointClassInformation>>() {
+                    });
+            List<RestCallFileInformation> restCallFiles = mapper.readValue(new File(EndpointParser.REST_CALL_PARSING_RESULT_PATH),
+                    new TypeReference<List<RestCallFileInformation>>() {
+                    });
+
+            List<UsedEndpoints> endpointsAndMatchingRestCalls = new ArrayList<>();
+            List<EndpointInformation> unusedEndpoints = new ArrayList<>();
+
+            Map<String, List<RestCallInformation>> restCallMap = new HashMap<>();
+
+            // Populate the map with rest calls
+            for (RestCallFileInformation restCallFile : restCallFiles) {
+                for (RestCallInformation restCall : restCallFile.restCalls()) {
+                    String restCallURI = restCall.buildComparableRestCallUri();
+                    restCallMap.computeIfAbsent(restCallURI, uri -> new ArrayList<>()).add(restCall);
```

Comment:
I think we can have different REST calls with the same URLs, e.g. one for creating something and one for deleting it. Should we handle this here?

Reply:
We do, but during the parsing process, an entry is made for each REST call, regardless of whether the URL is already used by another one. This is the reason why we are storing the restCalls of each URI in a List. That way, we can handle restCalls with the same URLs but different HTTP Methods.